### PR TITLE
Net-744: Tracker to its own package

### PR DESCRIPTION
**Pull request by juslesan**
Tracker is now its own npm package: `@streamr/network-tracker`

This PR makes a minimal effort to move tracker related logic and tests to its own package and create tracker specific NPM and Docker publications.

Further refactoring can be done in the future ie. the Network's `/connection` layer could be split between tracker and node related logic. For now the connection layer is kept as is (expect the webrtc related logic is moved to its own subdir) for convenience.

## Checklist

- [x] Has passing tests that demonstrate this change works
- [ ] Updated Changelog
- [x] Updated Documentation

### CODE REVIEW

File: packages/client/test/browser/browser.html [hunk 1]
```diff
@@ -232,6 +232,7 @@
         subscriberAuth
     })
 
+    console.info(StreamrClient.ConfigTest)
```
Comment (harbu) [inline packages/client/test/browser/browser.html]: Left in on purpose?

File: packages/network-tracker/package.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    "path-browserify": "^1.0.1",
+    "source-map-loader": "^3.0.0",
+    "streamr-test-utils": "^2.0.0",
+    "ts-jest": "^27.0.5",
+    "ts-loader": "^9.2.6",
+    "ts-node": "^9.1.1",
+    "typescript": "^4.5.2",
+    "webpack": "^5.64.1",
+    "webpack-cli": "^4.9.1"
+  },
```
Comment (harbu) [inline packages/network-tracker/package.json]: Are these all used in `network-tracker`?

**Review by harbu — APPROVED** [review]

File: packages/network-tracker/package.json (hunk 2 above)
Comment (juslesan) [reply packages/network-tracker/package.json]: There definitely seems to be too many devDependencies, I'll clean it up

File: packages/network-tracker/package.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "streamr-network-tracker",
```
Comment (kare) [inline packages/network-tracker/package.json]: Rename to `@streamr/network-tracker`? Can we rename this package with `@streamr` prefix and still have the binary as `streamr-network-tracker`?

File: packages/network-tracker/package.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    "test": "jest",
+    "test-integration": "jest test/integration/",
+    "test-types": "npm run check",
+    "test-unit": "jest test/unit",
+    "tracker": "node $NODE_DEBUG_OPTION bin/tracker.js"
+  },
+  "dependencies": {
+    "@streamr/slackbot": "^1.1.0",
+    "@types/compression": "^1.7.2",
+    "@types/morgan": "^1.9.3",
```
Comment (kare) [inline packages/network-tracker/package.json]: These belong to devDependencies

File: .github/workflows/broker.yml [hunk 5]
```diff
@@ -92,7 +92,7 @@ jobs:
         uses: docker/build-push-action@v2.10.0
         with:
           context: .
-          file: ./Dockerfile
+          file: ./Dockerfile-broker
```
Comment (kare) [inline .github/workflows/broker.yml]: Before merge I'd like experiment whether `Dockerfile` can be placed to respective `packages` directory. It didn't work some time ago, but might work now.

Comment (juslesan) [reply .github/workflows/broker.yml]: I tried this and it didn't really work, but there could be a way

File: .github/workflows/broker.yml [hunk 6]
```diff
@@ -206,7 +206,7 @@ jobs:
...
```
Comment (kare) [inline .github/workflows/broker.yml]:
```suggestion
          file: ./Dockerfile.broker
```

This way editors will recognise syntax highlighting correctly. Same thing for `Dockerfile-tracker`.

Comment (kare) [reply .github/workflows/broker.yml]: Renaming Dockerfile breaks `Makefile`!

File: Dockerfile-broker [hunk 7]
```diff
@@ -8,7 +9,7 @@ RUN --mount=type=cache,target=/root/.npm \
 	npm run bootstrap-pkg -- streamr-broker && \
 	npm run prune-pkg -- streamr-broker
 
-FROM --platform=$BUILDPLATFORM node:16.14-bullseye-slim
```
Comment (kare) [inline Dockerfile-broker]: This change needs to be reverted.

File: Dockerfile-broker [hunk 8]
```diff
@@ -1,5 +1,6 @@
-FROM --platform=$BUILDPLATFORM node:16.14-bullseye as build
```
Comment (kare) [inline Dockerfile-broker]: This change needs to be reverted.

File: Dockerfile-tracker [hunk 9]
```diff
@@ -0,0 +1,29 @@
+FROM node:16.14-bullseye as build
```
Comment (kare) [inline Dockerfile-tracker]: Needs `--platform=$BUILDPLATFORM`

File: Dockerfile-tracker [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+RUN npm install -g npm@8.1 --no-audit
+RUN npm config set \
+	unsafe-perm=true \
+	python="$(which python3)"
+COPY . .
+RUN --mount=type=cache,target=/root/.npm \
+	npm run bootstrap-pkg -- streamr-network-tracker && \
+	npm run prune-pkg -- streamr-network-tracker
+
+FROM node:16.14-bullseye-slim
```
Comment (kare) [inline Dockerfile-tracker]: Needs `--platform=$BUILDPLATFORM`

**Review by kare — CHANGES_REQUESTED** [review]
There's some minor issues:
- Mostly related to Dockerfiles
- Let's use `@streamr/` prefix for npm package 

And few major issues: 
- There's too much duplication in the CI build, that needs to be refactored
- Docker build needs some more thinking. This is a monorepo issue. Single `.dockerignore` won't work for two Docker builds.

File: packages/network-tracker/package.json [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  "name": "@streamr/network-tracker",
+  "version": "1.0.0",
+  "description": "Streamr Network Tracker package",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/streamr-dev/network-monorepo.git",
+    "directory": "packages/network-tracker"
+  },
+  "license": "STREAMR NETWORK OPEN SOURCE LICENSE",
+  "author": "Streamr Network AG <contact@streamr.network>",
```
Comment (kare) [inline packages/network-tracker/package.json:11]:
```suggestion
  "author": "Streamr Network AG <contact@streamr.network>",
  "bin": {
    "streamr-tracker": "bin/tracker.js"
  },
```

Completely untested...

File: Dockerfile.broker [hunk 12]
```diff
@@ -5,6 +5,7 @@ RUN npm config set \
 	python="$(which python3)"
 COPY . .
 RUN --mount=type=cache,target=/root/.npm \
+    npm run bootstrap-pkg -- streamr-client && \
```
Comment (kare) [inline Dockerfile.broker]: indent with tabs

File: .github/workflows/docker-build.yml [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+    secrets:
+      dockerhub_username:
+        description: 'Required if the docker image is pushed'
+        required: false
+      dockerhub_token:
+        required: false
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
```
Comment (kare) [inline .github/workflows/docker-build.yml:34]: Could this be `self-hosted`?

**Review by kare — APPROVED** [review]
LGTM, couple of comment/question.

File: .github/workflows/docker-build.yml (hunk 13 above)
Comment (juslesan) [reply .github/workflows/docker-build.yml:34]: It seems to take really long get self-hosted builds running. Could be temporary but it is quite annoying

Comment (kare) [reply .github/workflows/docker-build.yml:34]: Ok, good to know. I'll check what's up with the machine

Comment (kare) [reply .github/workflows/docker-build.yml:34]: Killed a stuck/zombie process, updated and rebooted. It should work now faster.

File: .github/workflows/docker-build.yml [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+        run: ${{inputs.test_script}}
+
+      - name: Build & Push
+        if: ${{inputs.push_image}}
+        uses: docker/build-push-action@v2.10.0
+        with:
+          context: .
+          file: ${{inputs.docker_file}}
+          platforms: ${{inputs.build_platforms}}
+          push: true
```
Comment (kare) [inline .github/workflows/docker-build.yml:107]: `${{inputs.push_image}}`?

File: .github/workflows/docker-build.yml [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+        if: ${{inputs.push_image}}
+        with:
+          images: ${{inputs.image_name}}
+          tags: |
+            type=sha
+            type=ref,event=branch
+            type=semver,pattern=v{{version}}
+            type=raw,value=dev,enable=${{ inputs.test_success }}
+            type=raw,value=debug,enable=${{ !inputs.test_success }}
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) && startsWith(github.ref, 'refs/tags/v') }}
```
Comment (kare) [inline .github/workflows/docker-build.yml:72]:
```suggestion
            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', github.event.repository.default_branch) && startsWith(github.ref, 'refs/tags/v') }}
            type=semver,pattern={{major}}.{{minor}},enable=${{ !startsWith(github.ref, 'refs/tags/v0.0.') }}
            type=semver,pattern={{major}},enable=${{ !startsWith(github.ref, 'refs/tags/v0.') }}
```

Comment (kare) [reply .github/workflows/docker-build.yml:72]: I'm adding two last lines, not changing the first one.

Pushes Docker Hub tags `1` or `1.1`, but not major `0` or `0.1`

**Pull request merged**